elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -282,6 +320,171 @@ private void addField(Field field, EntityProjection parentProjection) {
         }
     }
 
+    /**
+     * Construct Elide {@link Pagination}, {@link Sorting}, {@link Attribute} from GraphQL {@link Argument} and
+     * add it to the {@link EntityProjection}.
+     *
+     * @param argument graphQL argument
+     * @param entityProjection projection that has this argument
+     */
+    private void addArgument(Argument argument, EntityProjection entityProjection) {
+        String argumentName = argument.getName();
+
+        if (isPaginationArgument(argumentName)) {
+            addPagination(argument, entityProjection);
+        } else if (isSortingArgument(argumentName)) {
+            addSorting(argument, entityProjection);
+        } else if (ModelBuilder.ARGUMENT_FILTER.equals(argumentName)) {
+            addFilter(argument, entityProjection);
+        } else if (!ModelBuilder.ARGUMENT_OPERATION.equals(argumentName)
+                && !(ModelBuilder.ARGUMENT_IDS.equals(argumentName))
+                && !(ModelBuilder.ARGUMENT_DATA.equals(argumentName))) {
+            addAttributeArgument(argument, entityProjection);
+        }
+    }
+
+    /**
+     * Returns whether or not a GraphQL argument name corresponding to a pagination argument.
+     *
+     * @param argumentName Name key of the GraphQL argument
+     *
+     * @return {@code true} if the name equals to {@link ModelBuilder#ARGUMENT_FIRST} or
+     * {@link ModelBuilder#ARGUMENT_AFTER}
+     */
+    private static boolean isPaginationArgument(String argumentName) {
+        return ModelBuilder.ARGUMENT_FIRST.equals(argumentName) || ModelBuilder.ARGUMENT_AFTER.equals(argumentName);
+    }
+
+    /**
+     * Creates a {@link Pagination} object from pagination GraphQL argument and attaches it to the
+     * {@link EntityProjection}.
+     *
+     * @param argument graphQL argument
+     * @param entityProjection projection that has this argument
+     */
+    private void addPagination(Argument argument, EntityProjection entityProjection) {
+        Pagination pagination = entityProjection.getPagination() == null
+                ? Pagination.getDefaultPagination(elideSettings)
+                : entityProjection.getPagination();
+
+        Object argumentValue = variableResolver.resolveValue(argument.getValue());
+        int value = argumentValue instanceof BigInteger
```

Comment:
What if the value is an integer instead of a big integer?  Elide has a library called CoerceUtil.. Maybe you should use that instead.

Reply:
The resolver would only return `BigInteger`